Apologies for the lack of details, it turns out the paths are slightly different. Here are the errors.  
Upon loading NVDA and attempting to load add-ons, application dictionary, Rui's add-on, gives this.
```
ERROR - globalPluginHandler.listPlugins (21:34:51.307) - MainThread (22376):
Error importing global plugin applicationDictionary
Traceback (most recent call last):
  File "globalPluginHandler.pyc", line 23, in listPlugins
  File "importlib\__init__.pyc", line 127, in import_module
  File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "C:\Users\Colton\AppData\Roaming\nvda\addons\applicationDictionary\globalPlugins\applicationDictionary\__init__.py", line 82, in <module>
    appDictsPath = os.path.abspath(os.path.join(speechDictHandler.speechDictsPath, "appDicts"))
AttributeError: module 'speechDictHandler' has no attribute 'speechDictsPath'
```
And upon switching config profiles, enhanced dictionaries, which I got from an external source and is available [here](https://github.com/marlon-sousa/EnhancedDictionaries), gives this.
```
ERROR - extensionPoints.Action.notify (21:35:12.891) - MainThread (22376):
Error running handler <function _handlePostConfigProfileSwitch at 0x17975300> for <extensionPoints.Action object at 0x02F23510>
Traceback (most recent call last):
  File "extensionPoints\__init__.pyc", line 55, in notify
  File "extensionPoints\util.pyc", line 216, in callWithSupportedKwargs
  File "C:\Users\Colton\AppData\Roaming\nvda\addons\EnhancedDictionaries\globalPlugins\EnhancedDictionaries\__init__.py", line 18, in _handlePostConfigProfileSwitch
    dictHelper.reloadDictionaries()
  File "C:\Users\Colton\AppData\Roaming\nvda\addons\EnhancedDictionaries\globalPlugins\EnhancedDictionaries\dictHelper.py", line 40, in reloadDictionaries
    loadProfileDict()
  File "C:\Users\Colton\AppData\Roaming\nvda\addons\EnhancedDictionaries\globalPlugins\EnhancedDictionaries\dictHelper.py", line 93, in loadProfileDict
    if _hasDictionaryProfile(profile.name, "default.dic"):
  File "C:\Users\Colton\AppData\Roaming\nvda\addons\EnhancedDictionaries\globalPlugins\EnhancedDictionaries\dictHelper.py", line 128, in _hasDictionaryProfile
    return os.path.exists(os.path.join(dictFormatUpgrade.speechDictsPath, profileName or "", dictionaryName))
AttributeError: module 'speechDictHandler.dictFormatUpgrade' has no attribute 'speechDictsPath'
```
Seems strange that the second one attempts to be looking for the same path in a different place? I'm unsure how to actually test if this PR works myself without it being merged.

--- a/source/speechDictHandler/__init__.py
+++ b/source/speechDictHandler/__init__.py
@@ -157,6 +157,5 @@ It handles case when the synthesizer doesn't support voice setting.
 		baseName = dictFormatUpgrade.createVoiceDictFileName(synth.name, voice)
 	else:
 		baseName=r"{synth}.dic".format(synth=synth.name)
-	voiceDictsPath = dictFormatUpgrade.voiceDictsPath
-	fileName= os.path.join(voiceDictsPath, synth.name, baseName)
+	fileName = os.path.join(WritePaths.voiceDictsDir, synth.name, baseName)
 	dictionaries["voice"].load(fileName)

--- a/source/speechDictHandler/dictFormatUpgrade.py
+++ b/source/speechDictHandler/dictFormatUpgrade.py
@@ -1,12 +1,13 @@
 # -*- coding: UTF-8 -*-
 # A part of NonVisual Desktop Access (NVDA)
-# Copyright (C) 2017 NV Access Limited
+# Copyright (C) 2017-2023 NV Access Limited
 # This file may be used under the terms of the GNU General Public License, version 2 or later.
 # For more details see: https://www.gnu.org/licenses/gpl-2.0.html
 
 """Upgrade speech dict files
 """
 
+from typing import Any
 import globalVars
 import os
 import api
@@ -14,8 +15,38 @@ import glob
 from logHandler import log
 from NVDAState import WritePaths
 
-voiceDictsPath = WritePaths.voiceDictsDir
-voiceDictsBackupPath = WritePaths.voiceDictsBackupDir
+
+def __getattr__(attrName: str) -> Any:
+	"""Module level `__getattr__` used to preserve backward compatibility.
+	"""
+	import NVDAState
+	if NVDAState._allowDeprecatedAPI():
+		if attrName == "speechDictsPath":
+			log.warning(
+				"speechDictHandler.dictFormatUpgrade.speechDictsPath is deprecated, "
+				"instead use NVDAState.WritePaths.speechDictsDir",
+				stack_info=True
+			)
+			return WritePaths.speechDictsDir
+
+		if attrName == "voiceDictsPath":
+			log.warning(
+				"speechDictHandler.dictFormatUpgrade.voiceDictsPath is deprecated, "
+				"instead use NVDAState.WritePaths.voiceDictsDir",
+				stack_info=True
+			)
+			return WritePaths.voiceDictsDir
+
+		if attrName == "voiceDictsBackupPath":
+			log.warning(
+				"speechDictHandler.dictFormatUpgrade.voiceDictsBackupPath is deprecated, "
+				"instead use NVDAState.WritePaths.voiceDictsBackupDir",
+				stack_info=True
+			)
+			return WritePaths.voiceDictsBackupDir
+
+	raise AttributeError(f"module {repr(__name__)} has no attribute {repr(attrName)}")
+
 
 def createVoiceDictFileName(synthName, voiceName):
 	""" Creates a filename used for the voice dictionary files.
@@ -56,12 +87,12 @@ def _doSynthVoiceDictBackupAndMove(synthName, oldFileNameToNewFileNameList=None)
 	"""
 	import shutil
 	
-	if not os.path.isdir(voiceDictsPath):
-		os.makedirs(voiceDictsPath)
-	if not os.path.isdir(voiceDictsBackupPath):
-		os.makedirs(voiceDictsBackupPath)
+	if not os.path.isdir(WritePaths.voiceDictsDir):
+		os.makedirs(WritePaths.voiceDictsDir)
+	if not os.path.isdir(WritePaths.voiceDictsBackupDir):
+		os.makedirs(WritePaths.voiceDictsBackupDir)
 	
-	newDictPath = os.path.join(voiceDictsPath,synthName)
+	newDictPath = os.path.join(WritePaths.voiceDictsDir, synthName)
 	needsUpgrade = not os.path.isdir(newDictPath)
 	if needsUpgrade:
 		log.info("Upgrading voice dictionaries for %s"%synthName)
@@ -82,7 +113,7 @@ def _doSynthVoiceDictBackupAndMove(synthName, oldFileNameToNewFileNameList=None)
 			log.debug("processing file: %s" % actualPath)
 			# files will be copied here before we modify them so as to avoid
 			# any data loss.
-			shutil.copy(actualPath, voiceDictsBackupPath)
+			shutil.copy(actualPath, WritePaths.voiceDictsBackupDir)
 			
 			actualBasename = os.path.basename(actualPath)
 			log.debug("basename: %s" % actualBasename)

--- a/user_docs/en/changes.t2t
+++ b/user_docs/en/changes.t2t
@@ -152,6 +152,8 @@ It was introduced in NVDA 2023.1 and was never meant to be part of the public AP
 Until removal, it behaves as a no-op, i.e. a context manager yielding nothing. (#14924)
 - ``gui.MainFrame.onAddonsManagerCommand`` is deprecated, use ``gui.MainFrame.onAddonStoreCommand`` instead. (#13985)
 - ``speechDictHandler.speechDictVars.speechDictsPath`` is deprecated, use ``NVDAState.WritePaths.speechDictsDir`` instead. (#15021)
+- Importing ``voiceDictsPath`` and ``voiceDictsBackupPath`` from ``speechDictHandler.dictFormatUpgrade`` is deprecated.
+Instead use ``WritePaths.voiceDictsDir`` and ``WritePaths.voiceDictsBackupDir`` from ``NVDAState``. (#15048)
 -
 
 = 2023.1 =

